### PR TITLE
feat(models): add support for generic platform resources

### PR DESCRIPTION
**Pull request by shirshanka**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [x] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-models/src/main/pegasus/com/linkedin/platformresource/PlatformResourceInfo.pdl [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    * Intended as a loose specifier of the generic type of the resource.
+    * Producer is not forced to conform to a specific set of symbols for
+    * resource types.
+    * The @PlatformResourceType enumeration offers a paved path for agreed upon
+    * common terms, but is not required to be followed.
+    * Example values could be: conversation, user, grant, etc.
+    * Resource types are indexed for ease of access. 
+    * e.g. Get me all platform resources of type user for the platform looker
+   */
+   @Searchable = {
```
Comment (david-leifker) [inline metadata-models/src/main/pegasus/com/linkedin/platformresource/PlatformResourceInfo.pdl:25]: While these are not enabled for general search at the moment, if they ever are we probably want to consider which of these searchable fields should be queried by default from the main search bar. Ones that should not be included should be marked with `"queryByDefault": false`. If this entity will never be involved in main search then no need.

Comment (shirshanka) [reply metadata-models/src/main/pegasus/com/linkedin/platformresource/PlatformResourceInfo.pdl:25]: The plan is to NOT make these resources visible in the Search UI ... are you recommending I add queryByDefault: false to make sure this never happens accidentally?

Comment (david-leifker) [reply metadata-models/src/main/pegasus/com/linkedin/platformresource/PlatformResourceInfo.pdl:25]: No need. If this entity is ever added to entity types to search by default in the graphql layer, I've created a test to ensure this is considered.

**Review by david-leifker — APPROVED** [review]

**Review by david-leifker — CHANGES_REQUESTED** [review]
```
56,18: Type not found: SerializedValue

	at com.linkedin.data.schema.generator.AbstractGenerator.parseSources(AbstractGenerator.java:157)
	at com.linkedin.data.avro.generator.AvroSchemaGenerator.generate(AvroSchemaGenerator.java:191)
	at com.linkedin.data.avro.generator.AvroSchemaGenerator.run(AvroSchemaGenerator.java:165)
	at com.linkedin.data.avro.generator.AvroSchemaGenerator.main(AvroSchemaGenerator.java:123)
```

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**